src/index.js

```diff
@@ -329,6 +347,63 @@ export default class AuthnWidget {
     }
   }
 
+  postEmptyAuthentication() {
+    this.store.dispatch('POST_FLOW', 'authenticate', '{}');
+  }
+
+  registerMfaEventHandler() {
+    Array.from(document.querySelectorAll('[data-mfa-selection]'))
+      .forEach(element => element.addEventListener('click', this.handleMfaDeviceSelection));
+  }
+
+  handleMfaDeviceSelection(evt) {
+    evt.preventDefault();
+    let source = evt.currentTarget;
+    if (source) {
+      let deviceId = source.dataset['mfaSelection'];
+      console.log(deviceId)
+      let data = {
+        "deviceRef": {
+          "id": deviceId
+        }
+      };
+      this.store.dispatch('POST_FLOW', "selectDevice", JSON.stringify(data));
+    } else {
+      console.log("ERROR - Unable to dispatch device selection as the target was null");
+    }
+  }
+
+  registerMfaChangeDeviceEventHandler() {
+    Array.from(document.querySelectorAll('#changeDevice'))
+      .forEach(element => element.addEventListener('click', this.handleMfaDeviceChange));
+  }
+
+  async handleMfaDeviceChange(evt) {
+    evt.preventDefault();
+    let state = await this.store.getState();
+    state.status = 'DEVICE_SELECTION_REQUIRED';
+    this.render(this.store.getPreviousStore(), state);
```

Comment:
This is a little bit worrisome to me to force the widget to be out of sync with the state on the server. Is this how it works in the current kit? Seems like there should be an action associated with the back button that will transition the state back to DEVICE_SELECTION_REQUIRED

Reply:
Yeah, I'm not a big fan of doing this either... The adapter templates currently changes state back to `DEVICE_SELECTION_REQUIRED`, but P1C does not and apparently that's how it's going to be in the next adapter release as well so behaviour is the same.

Reply:
Can we comment on that defect that the widget should be updated here if the adapter changes the way it works as well?